src/SBInterpolatedImage.cpp

```diff
@@ -895,31 +881,61 @@ namespace galsim {
         }
         _flux = kValue(Position<double>(0.,0.)).real();
         dbg<<"flux = "<<_flux<<std::endl;
+        setCentroid();
+    }
+
+    void SBInterpolatedKImage::SBInterpolatedKImageImpl::setCentroid() const {
```

Comment:
I went ahead and made a `_InterpolatedKImage` function that creates a `InterpolatedKImage` object without the sanity checks.  Using it in `ChromaticRealGalaxy` brings that 2.5% above down to 0.6%.